tests/asgi/tests.py

```diff
@@ -425,11 +440,25 @@ class TestASGIHandler(ASGIHandler):
             await communicator.receive_output()
         await communicator.wait()
         self.assertIs(view_did_cancel, True)
+        # Assert exactly one call to on_finished handler
+        self.assertEqual(len(finished_threads), 1)
+        # Assert it was NOT on the async thread
```

Comment:
To add more explanation, docstrings/comments should state the expected behavior and omit prefixes like "Assert" since all assertions assert things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).